pkgs/applications/misc/gramps/default.nix

```diff
@@ -21,47 +61,30 @@ in buildPythonApplication rec {
     # Ghostscript support
     ++ lib.optional enableGhostscript ghostscript
   ;
+  propagatedBuildInputs = with pythonPackages; [ bsddb3 PyICU pygobject3 pycairo ];
+  checkInputs = with pythonPackages; [ glibcLocales jsonschema lxml mock ];
 
-  src = fetchFromGitHub {
-    owner = "gramps-project";
-    repo = "gramps";
-    rev = "v${version}";
-    sha256 = "00358nzyw686ypqv45imc5k9frcqnhla0hpx9ynna3iy6iz5006x";
-  };
-
-  pythonPath = with pythonPackages; [ bsddb3 PyICU pygobject3 pycairo ];
-
-  # Same installPhase as in buildPythonApplication but without --old-and-unmanageble
-  # install flag.
-  installPhase = ''
-    runHook preInstall
-
-    mkdir -p "$out/lib/${python.libPrefix}/site-packages"
-
-    export PYTHONPATH="$out/lib/${python.libPrefix}/site-packages:$PYTHONPATH"
-
-    ${python}/bin/${python.executable} setup.py install \
-      --install-lib=$out/lib/${python.libPrefix}/site-packages \
-      --prefix="$out"
-
-    eapth="$out/lib/${python.libPrefix}"/site-packages/easy-install.pth
-    if [ -e "$eapth" ]; then
-        # move colliding easy_install.pth to specifically named one
-        mv "$eapth" $(dirname "$eapth")/${pname}-${version}.pth
-    fi
-
-    rm -f "$out/lib/${python.libPrefix}"/site-packages/site.py*
-
-    runHook postInstall
+  preCheck = ''
+    # HOME must be /tmp rather than mktemp -d in order to skip a specific test
+    # https://github.com/gramps-project/gramps/pull/1347
+    export HOME="/tmp"
```

Comment:
why do we have a `preCheck` Phase if `doCheck` is false?

Reply:
Because this is what makes the tests work when they can be enabled. Its just the one patch from a future version that was needed broke the tests again.

It'll be helpful when they're turned back on hopefully next release

Reply:
Ok, maybe I missed it, but it wasn't clear to me. Maybe add a comment about it, so the next person re-enables the tests on a package update?

Reply:
There's a comment above the doCheck and another above the patch it refers to :)


Also the person updating it will almost certainly be me. There was no maintainer anymore (so I added myself)

Reply:
ah, perfect, I missed that